core/src/forward_packet_batches_by_accounts.rs

```diff
@@ -183,140 +163,159 @@ mod tests {
     use {
         super::*,
         crate::unprocessed_packet_batches::DeserializedPacket,
-        solana_runtime::transaction_priority_details::TransactionPriorityDetails,
+        solana_runtime::{
+            bank::Bank,
+            genesis_utils::{create_genesis_config, GenesisConfigInfo},
+            transaction_priority_details::TransactionPriorityDetails,
+        },
         solana_sdk::{
-            feature_set::FeatureSet, hash::Hash, signature::Keypair, system_transaction,
-            transaction::SimpleAddressLoader,
+            feature_set::FeatureSet, hash::Hash, pubkey::Pubkey, signature::Keypair,
+            system_transaction,
         },
         std::sync::Arc,
     };
 
-    fn build_deserialized_packet_for_test(
+    fn test_setup() -> (Keypair, Hash) {
+        solana_logger::setup();
+        let GenesisConfigInfo {
+            genesis_config,
+            mint_keypair,
+            ..
+        } = create_genesis_config(10);
+        let bank = Arc::new(Bank::new_no_wallclock_throttle_for_tests(&genesis_config));
+        let start_hash = bank.last_blockhash();
+        (mint_keypair, start_hash)
+    }
+
+    /// build test transaction, return corresponding sanitized_transaction and deserialized_packet,
+    /// and the batch limit_ratio that would only allow one transaction per bucket.
+    fn build_test_transaction_and_packet(
         priority: u64,
         write_to_account: &Pubkey,
-        compute_unit_limit: u64,
-    ) -> DeserializedPacket {
-        let tx =
-            system_transaction::transfer(&Keypair::new(), write_to_account, 1, Hash::new_unique());
-        let packet = Packet::from_data(None, tx).unwrap();
-        DeserializedPacket::new_with_priority_details(
-            packet,
+    ) -> (SanitizedTransaction, DeserializedPacket, u32) {
+        let (mint_keypair, start_hash) = test_setup();
+        let transaction =
+            system_transaction::transfer(&mint_keypair, write_to_account, 2, start_hash);
+        let sanitized_transaction =
+            SanitizedTransaction::from_transaction_for_tests(transaction.clone());
+        let tx_cost = CostModel::calculate_cost(&sanitized_transaction, &FeatureSet::all_enabled());
+        let cost = tx_cost.sum();
+        let deserialized_packet = DeserializedPacket::new_with_priority_details(
+            Packet::from_data(None, transaction).unwrap(),
             TransactionPriorityDetails {
                 priority,
-                compute_unit_limit,
+                compute_unit_limit: cost,
             },
         )
-        .unwrap()
+        .unwrap();
+
+        // set limit ratio so each batch can only have one test transaction,
+        // eg ratio = floor(MAX_WRITABLE_ACCOUNT_UNITS- / tx_cost.sum())
```

Comment:
either fix the comment (trailling `-`) or just remove it. I don't think it adds much since the equation is simply the coded line below.

Reply:
good point, removed the line